repose-aggregator/commons/commons-utilities/src/test/scala/org/openrepose/commons/utils/opentracing/httpclient/ReposeTracingRequestInterceptorTest.scala

```diff
@@ -19,81 +19,146 @@
+    when(spanBuilder.start()).thenReturn(span)
+    when(tracer.buildSpan(s"${requestLine.getMethod} $redactedPath")).thenReturn(spanBuilder)
+  }
 
   describe("testOnSpanStarted") {
     it("no headers") {
-      val httpRequest = mock[HttpRequest]
-      val httpContext = mock[HttpContext]
-      val tracer = mock[Tracer]
-      val scopeManager = mock[ScopeManager]
-      val scope = mock[Scope]
-      val span = mock[Span]
+      val httpRequestInterceptor = new SurrogateReposeTracingRequestInterceptor(tracer, "1.two.III", uriRedactionService)
+
+      httpRequestInterceptor.onSpanStarted(span, httpRequest, httpContext)
+
+      verify(span).setTag(ReposeVersion, "1.two.III")
+      verify(uriRedactionService).redact(requestLine.getUri)
+      verify(span).setTag(HTTP_URL.toString, redactedPath)
+    }
 
-      when(scope.span()).thenReturn(span)
-      when(scopeManager.active()).thenReturn(scope)
-      when(tracer.scopeManager()).thenReturn(scopeManager)
+    it("with request header") {
+      when(httpRequest.getFirstHeader(any())).thenReturn(new BasicHeader(REQUEST_ID, "1234"))
 
-      val tracingRequestInterceptor = new ReposeTracingRequestInterceptor(tracer, "1.two.III")
+      val httpRequestInterceptor = new SurrogateReposeTracingRequestInterceptor(tracer, "1.two.III", uriRedactionService)
 
-      tracingRequestInterceptor.process(httpRequest, httpContext)
+      httpRequestInterceptor.onSpanStarted(span, httpRequest, httpContext)
 
-      verify(span).setTag(ReposeTags.ReposeVersion, "1.two.III")
+      verify(span).setTag(REQUEST_ID, "1234")
+      verify(uriRedactionService).redact(requestLine.getUri)
+      verify(span).setTag(HTTP_URL.toString, redactedPath)
     }
 
-    it("with request header") {
-      val httpRequest = mock[HttpRequest]
-      val httpContext = mock[HttpContext]
-      val tracer = mock[Tracer]
-      val scopeManager = mock[ScopeManager]
-      val scope = mock[Scope]
-      val span = mock[Span]
+    it("with via header") {
+      when(httpRequest.getFirstHeader(any())).thenReturn(new BasicHeader(VIA, "1234"))
 
-      when(httpRequest.getFirstHeader(any())).thenReturn(new BasicHeader(CommonHttpHeader.REQUEST_ID, "1234"))
-      when(scope.span()).thenReturn(span)
-      when(scopeManager.active()).thenReturn(scope)
-      when(tracer.scopeManager()).thenReturn(scopeManager)
+      val httpRequestInterceptor = new SurrogateReposeTracingRequestInterceptor(tracer, "1.two.III", uriRedactionService)
 
-      val jaegerRequestInterceptor = new ReposeTracingRequestInterceptor(tracer, "1.two.III")
+      httpRequestInterceptor.onSpanStarted(span, httpRequest, httpContext)
 
-      jaegerRequestInterceptor.process(httpRequest, httpContext)
+      verify(span).setTag(VIA, "1234")
+      verify(uriRedactionService).redact(requestLine.getUri)
+      verify(span).setTag(HTTP_URL.toString, redactedPath)
+    }
+  }
+
+  describe("testGetOperationName") {
```

Comment:
Ah. When I made my comment about testing `getOperationName` and `onSpanStarted`, I guess I didn't realize that they are protected methods. So you were probably right that they don't strictly need to be tested, and thus, we may not strictly need the anonymous subclass that grants access to them. Also, as you pointed out, testing them leads to some duplication.

Reply:
I'm good either way. The subclass is only in the test, so I'm not that concerned with it. Since it is only two methods and they have very specific behavior, I'm not opposed to testing them. If anything, then removing the dupes in favor of just making sure the two tested protected methods are called by the one public.

Reply:
I guess I'm also good with it either way.